lib/ruby_indexer/lib/ruby_indexer/index.rb

```diff
@@ -150,17 +150,42 @@ def method_completion_candidates(name, receiver_name)
       ancestors = linearized_ancestors_of(receiver_name)
 
       candidates = name ? prefix_search(name).flatten : @entries.values.flatten
-      candidates.filter_map do |entry|
-        case entry
-        when Entry::Member, Entry::MethodAlias
-          entry if ancestors.any?(entry.owner&.name)
-        when Entry::UnresolvedMethodAlias
-          if ancestors.any?(entry.owner&.name)
-            resolved_alias = resolve_method_alias(entry, receiver_name)
-            resolved_alias if resolved_alias.is_a?(Entry::MethodAlias)
-          end
+      completion_items = candidates.each_with_object({}) do |entry, hash|
```

Comment:
We could, but I believe that will be slower because we'll be looping through the rather large array of all methods that matched the prefixed another time.